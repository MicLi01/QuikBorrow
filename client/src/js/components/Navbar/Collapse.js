import React, { Component } from 'react';
import Collapse from '@material-ui/core/Collapse';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import Typography from '@material-ui/core/Typography';

// https://github.com/mui-org/material-ui/blob/master/docs/src/pages/components/cards/RecipeReviewCard.js

export default function Collapsed() {
    const [expanded, setExpanded] = React.useState(false);

    const handleExpandClick = () => {
        setExpanded(!expanded);
    };
    return (
        <div>
            <IconButton
          onClick={handleExpandClick}
          aria-expanded={expanded}
          aria-label="show more"
            >
          <ExpandMoreIcon />
        </IconButton>
      <Collapse in={expanded} timeout="auto" unmountOnExit>
          <Typography gutterBottom variant="h5"> Edit Profile </Typography>

        </Collapse>
        </div>
    )
}
