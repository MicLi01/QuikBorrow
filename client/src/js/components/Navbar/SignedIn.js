import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { logoutUser } from '../../actions';
import { Toolbar, Avatar } from '@material-ui/core';
import styled from 'styled-components'
import AddItemModal from '../AddItemModal/AddItemModal';

const UnstyledLink = styled(Link)`
    color: inherit;
    text-decoration: inherit;
    margin-left: 20px;
`

const SignedIn = (props) => {
    const onClick = e => {
        e.preventDefault();
        props.logoutUser();
    };
    const userProfileId = props.id 

    const [ openModal, setOpenModal ] = useState(false)

    return (
        <Toolbar>
            <AddItemModal open={openModal} onClose={e => setOpenModal(false)} />
            
            <UnstyledLink onClick={e => setOpenModal(true)}>Add Item</UnstyledLink>
            <UnstyledLink to='/products'>Products</UnstyledLink>
            <UnstyledLink to={ '/profile/'+userProfileId}><Avatar src='https://avatars1.githubusercontent.com/u/619960?s=460&v=4'/> </UnstyledLink>
            <UnstyledLink to='#' onClick={onClick}>Sign Out</UnstyledLink>
        </Toolbar>
    )
}

export default connect(null, { logoutUser })(SignedIn);
