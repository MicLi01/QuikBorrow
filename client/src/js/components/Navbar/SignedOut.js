import React from 'react';
import { Link } from 'react-router-dom';
import { Toolbar } from '@material-ui/core';
import styled from 'styled-components'

const UnstyledLink = styled(Link)`
    color: inherit;
    text-decoration: inherit;
    margin-left: 20px;
`

const SignedOut = () => {
    return (
        <Toolbar>
            <UnstyledLink to='/login'>Login</UnstyledLink>
            <UnstyledLink to='/register'>Register</UnstyledLink>
        </Toolbar>
    )
}

export default SignedOut;