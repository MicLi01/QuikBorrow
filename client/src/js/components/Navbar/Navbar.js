import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import styled from 'styled-components';
import SignedIn from './SignedIn';
import SignedOut from './SignedOut';
import { AppBar, Toolbar, IconButton, Typography } from '@material-ui/core';
import teal from '@material-ui/core/colors/teal'

const UnstyledLink = styled(Link)`
    color: inherit;
    text-decoration: inherit;
`

export class Navbar extends React.Component {
    render() {
        return (
            <AppBar position='fixed' style={{background: teal[500], boxShadow: 'none', color: 'white'}}>
                <Toolbar>
                    <IconButton edge="start" color="inherit" aria-label="menu">
                    </IconButton>
                        <Typography variant="h5" component='h5' style={{flexGrow: 1}}>
                            <UnstyledLink to='/' className='brand-logo'>
                            QuikBorrow
                            </UnstyledLink>
                        </Typography>
                    {this.props.auth.isAuthenticated ? <SignedIn name={this.props.auth.user.name} /> : <SignedOut />}
                </Toolbar>
            </AppBar>
        )
    }
}

function mapStateToProps(state) {
    return { auth : state.auth }
}

export default connect(
    mapStateToProps
)(Navbar);
