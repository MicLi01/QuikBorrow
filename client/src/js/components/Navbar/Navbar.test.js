import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import { Navbar } from './Navbar';

describe('Navbar', () => {
    it('should render a navbar with SignedIn links', () => {
        const component = renderer.create(
			<MemoryRouter>
				<Navbar auth = {true} />
			</MemoryRouter>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
    });

    it('should render a navbar with SignedOut links', () => {
        const component = renderer.create(
			<MemoryRouter>
				<Navbar auth = {false} />
			</MemoryRouter>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
    });
});