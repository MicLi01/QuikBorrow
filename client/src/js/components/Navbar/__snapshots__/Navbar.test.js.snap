// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render a navbar with SignedIn links 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
  style={
    Object {
      "background": "#009688",
      "boxShadow": "none",
      "color": "white",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      />
    </button>
    <h5
      className="MuiTypography-root MuiTypography-h5"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <a
        className="sc-htpNat hPRntI brand-logo"
        href="/"
        onClick={[Function]}
      >
        QuikBorrow
      </a>
    </h5>
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        className="sc-bwzfXH jeMjYc"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        className="sc-bwzfXH jeMjYc"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
</header>
`;

exports[`Navbar should render a navbar with SignedOut links 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed"
  style={
    Object {
      "background": "#009688",
      "boxShadow": "none",
      "color": "white",
    }
  }
>
  <div
    className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
  >
    <button
      aria-label="menu"
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-edgeStart"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      />
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <h5
      className="MuiTypography-root MuiTypography-h5"
      style={
        Object {
          "flexGrow": 1,
        }
      }
    >
      <a
        className="sc-htpNat hPRntI brand-logo"
        href="/"
        onClick={[Function]}
      >
        QuikBorrow
      </a>
    </h5>
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        className="sc-bwzfXH jeMjYc"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
      <a
        className="sc-bwzfXH jeMjYc"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </div>
  </div>
</header>
`;
